examples/api-samples/src/browser/api-samples-frontend-module.ts

```diff
@@ -24,13 +24,28 @@ import { bindSampleFileWatching } from './file-watching/sample-file-watching-con
 import { bindVSXCommand } from './vsx/sample-vsx-command-contribution';
 
 import '../../src/browser/style/branding.css';
+import { workspacePreferenceSchema, WorkspacePreferencesSchemaProvider } from '@theia/workspace/lib/browser/workspace-preferences';
+import { PreferenceSchema } from '@theia/core/lib/browser';
 
-export default new ContainerModule(bind => {
+export default new ContainerModule((bind, unbind, isBound, rebind) => {
     bindDynamicLabelProvider(bind);
     bindSampleUnclosableView(bind);
     bindSampleOutputChannelWithSeverity(bind);
     bindSampleMenu(bind);
     bindSampleFileWatching(bind);
     bindVSXCommand(bind);
     bindSampleFilteredCommandContribution(bind);
+    bind(TestWorkspacePreferencesSchemaProvider).toSelf().inSingletonScope();
+    rebind(WorkspacePreferencesSchemaProvider).toService(TestWorkspacePreferencesSchemaProvider);
 });
+
+@injectable()
+class TestWorkspacePreferencesSchemaProvider extends WorkspacePreferencesSchemaProvider {
+
+    get schema(): PreferenceSchema {
+        const schema = workspacePreferenceSchema;
+        delete schema.properties['workspace.preserveWindow'];
+        return schema;
+    }
+
+}
```

Comment:
Note: the following commit https://github.com/eclipse-theia/theia/commit/c5a7d712d21a7f89eb03c03244ac1b91f38d02a6 will be dropped (which includes this change). The change is present to illustrate that we can now extend/modify schemas.

Reply:
Hmm... If you are going to delete a field on an imported object like that you don't even need to rebind anything. That's basically monkey-patching: since you are mutating a shared reference it would update all other references as well.